File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    stack = []
    # 문자열 순회하면서 stack의 peek가 현재 문자이면 현재 문자 pop, 아니면 push
    for c in s:
        if stack[-1:] == [c]:
            stack.pop()
        else:
            stack.append(c)
    #print(0 if stack else 1)
    return 0 if stack else 1

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.*;
class Solution
{
    public int solution(String s)
    {   /*
        스택이 비거나 top과 다르면 stack에 넣는다.
        스택이 비지 않았고 top과 같으면 pop한다.
       
        문자열 순회한 뒤 비어있지 않으면 0 
        */
        Stack<Character> stack = new Stack();
        for(char ch: s.toCharArray()){
            if(!stack.isEmpty() && stack.peek() == ch){
                stack.pop();
            }
            else{
                stack.push(ch);
            }
        }
        if(!stack.isEmpty()){
            return 0;
        }
        return 1;
    }
}